### PR TITLE
Use scribe.LoggerSupport instead of scribe.Logger

### PR DESCRIPTION
**Pull request by olafurpg**
`LoggerSupport` is a minimal trait that contains all the methods use and
`scribe.Logger` is a case class with a lot of default parameter values.

### CODE REVIEW

File: benchmarks/src/main/scala/benchmarks/HotBaseProtocolMessageParserBench.scala [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-    val messages = BaseProtocolMessage.fromByteBuffers(
-      Observable(requests: _*).map(MessageWriter.write),
-      Logger("bench")
-    )
-    val s = TestScheduler()
-    val f = messages.runAsyncGetLast(s)
-    while (s.tickOne()) ()
-    Await.result(f, Duration("10s"))
-  }
-}
```
Comment (gabro) [inline benchmarks/src/main/scala/benchmarks/HotBaseProtocolMessageParserBench.scala:37]: This removal seems out of scope

Comment (olafurpg) [reply benchmarks/src/main/scala/benchmarks/HotBaseProtocolMessageParserBench.scala:37]: This is dead code that was neither tested not compiled as part of the project. I found it because it referenced scala-logging

**Comment by olafurpg** [comment]
Please merge and cut v0.2.0-M1 if all looks good, I gotta run for a quick errand

**Review by jvican — APPROVED** [review]
LGTM, thanks for the quick fix!

**Comment by jvican** [comment]
I'll be cutting 1.0.0-RC1 soon as discussed with Olaf offline.
